presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/PlanMatchPattern.java

```diff
@@ -187,26 +193,124 @@ public static PlanMatchPattern markDistinct(
                 Optional.of(new SymbolAlias(hashSymbol))));
     }
 
-    public static PlanMatchPattern window(
-            ExpectedValueProvider<WindowNode.Specification> specification,
-            List<ExpectedValueProvider<FunctionCall>> windowFunctions,
-            PlanMatchPattern source)
+    public static ExpectedValueProvider<WindowNode.Frame> windowFrame(
+            WindowFrame.Type type,
+            FrameBound.Type startType,
+            Optional<String> startValue,
+            FrameBound.Type endType,
+            Optional<String> endValue)
     {
-        PlanMatchPattern result = node(WindowNode.class, source).with(new WindowMatcher(specification));
-        windowFunctions.forEach(
-                function -> result.withAlias(Optional.empty(), new WindowFunctionMatcher(function)));
-        return result;
+        return new WindowFrameProvider(
+                type,
+                startType,
+                startValue.map(SymbolAlias::new),
+                endType,
+                endValue.map(SymbolAlias::new));
     }
 
-    public static PlanMatchPattern window(
-            ExpectedValueProvider<WindowNode.Specification> specification,
-            Map<String, ExpectedValueProvider<FunctionCall>> assignments,
-            PlanMatchPattern source)
+    public static PlanMatchPattern window(Consumer<WindowBuilder> windowBuilderConsumer, PlanMatchPattern source)
     {
-        PlanMatchPattern result = node(WindowNode.class, source).with(new WindowMatcher(specification));
-        assignments.entrySet().forEach(
-                assignment -> result.withAlias(assignment.getKey(), new WindowFunctionMatcher(assignment.getValue())));
-        return result;
+        WindowBuilder windowBuilder = new WindowBuilder(source);
+        windowBuilderConsumer.accept(windowBuilder);
+        return windowBuilder.build();
+    }
+
+    public static class WindowBuilder
```

Comment:
I moved it into WindowMatcher.Builder, and tried to get the access control on the member functions as narrow as possible.